pkg/endpoint/bpf.go

```diff
@@ -450,6 +450,18 @@ func (e *Endpoint) regenerateBPF(owner Owner, currentDir, nextDir string, regenC
 	proxyWaitGroup := completion.NewWaitGroup(completionCtx)
 	defer cancel()
 
+	// Keep track of the side-effects of the regeneration that need to be
+	// reverted in case of failure.
+	var revertStack RevertStack
+	defer func() {
+		if reterr != nil {
+			e.Logger().Error("Restoring endpoint state after BPF regeneration failed")
+			if err := e.Revert(&revertStack); err != nil {
```

Comment:
Maybe these changes is in the wrong commit? I don't see a `e.Push`

Reply:
@aanm I think this is just the framework part of it - with just this commit, there aren't any functional changes since nothing is pushed yet.

Reply:
Correct, this is built in multiple commits.